layouts/_default/baseof.html

```diff
@@ -2,6 +2,7 @@
 <html class="no-js" lang="{{ (string ( $.Site.Language) ) }}" dir="ltr">
 {{ $pageTitle := i18n .Params.title }}
 {{ $titleKey := .Params.title }}
+{{ $js := resources.Get "js/cds-app.js" | js.Build (dict "params" (dict "api" .Site.Params.lever_api_endpoint ) ) }}
```

Comment:
Do you know if there's a way to pass in all parameters to the build process so that we can add more env vars in the future without needing to redefine the `dict` and its keys?  And no is a totally legit answer - my Hugo knowledge is weak 😅 

My feeling is that it would look something like this:
```suggestion
{{ $js := resources.Get "js/cds-app.js" | js.Build (dict "params" .Site.Params ) }}
```

And then the only difference is in your Javascript above you'd do this:
```js
const endpoint = params.lever_api_endpoint;
```

Reply:
You're right, and spot on! This is indeed possible and is a good approach for future-proofing the code. Your suggested change is correct:

```html
{{ $js := resources.Get "js/cds-app.js" | js.Build (dict "params" .Site.Params) }}
```

This approach has many advantages:

1. Scalability: We can add new parameters to your `config.yaml` without needing to modify the Hugo template each time.

2. Flexibility: All site parameters are available in the JavaScript, allowing us to use them as needed without rebuilding the site for each new parameter.

3. Maintainability: The code is cleaner and requires less manual updating.

In the JavaScript, we would then access parameters like this:

```javascript
import * as params from '@params';

const endpoint = params.lever_api_endpoint;
// You can now easily use other params as well
const otherParam = params.some_other_param;
```

A few things to keep in mind:

1. Security: We need to be cautious about which parameters we expose to the client-side JavaScript. We have to avoid including sensitive information in `.Site.Params` that we don't want to be publicly accessible.

2. File Size: If we have a large number of parameters, this approach will include all of them in the JavaScript build, potentially increasing the file size slightly.

3. Type Safety: In a TypeScript environment, we might want to create type definitions for the params to get better IDE support and catch potential errors early.

Your suggestion is an excellent way to make the Hugo setup more flexible and easier to maintain in the long run. It's a common and recommended practice when working with Hugo and JavaScript together.

Reply:
Ah perfect and very good points about the things we'll need to consider as more params are added!